lib/system/freertos/irq.c

```diff
@@ -162,13 +166,12 @@ int metal_irq_register(int irq,
 }
 
 /* helper function for metal_irq_unregister() */
-static void metal_irq_delete_node(struct metal_list **node, void *p_to_free)
+static void metal_irq_delete_node(struct metal_list *node, void *p_to_free)
 {
 	unsigned int irq_flags_save;
 
-	*node = (*node)->prev;
 	irq_flags_save=metal_irq_save_disable();
-	metal_list_del((*node)->next);
```

Comment:
this now return a freed pointer, the foreach loop around this may break as it reference that pointer.